### PR TITLE
Add Catalogue pipeline storage Elastic Cloud cluster

### PR DESCRIPTION
**Pull request by kenoir**
This PR adds configuration for an Elastic Cloud cluster in terraform and AWS PrivateLink connection to the infrastructure critical stack.  

The intention is to reduce NAT Gateway costs and better manage infrastructure as code.

These changes add a new cluster, but do not attach it to a new pipeline, which should occur in an upcoming PR.

Part of: https://github.com/wellcomecollection/platform/issues/4949

### CODE REVIEW

File: infrastructure/critical/locals.tf [hunk 1]
```diff
@@ -7,6 +7,13 @@ locals {
   public_subnets_new  = local.catalogue_vpcs["catalogue_vpc_delta_public_subnets"]
   private_subnets_new = local.catalogue_vpcs["catalogue_vpc_delta_private_subnets"]
 
+  catalogue_pipeline_ec_vpce_domain      = "vpce.eu-west-1.aws.elastic-cloud.com"
+  catalogue_pipeline_ec_privatelink_host = "${ec_deployment.pipeline_storage.id}.${local.catalogue_pipeline_ec_vpce_domain}"
```
Comment (jamesgorrie) [inline infrastructure/critical/locals.tf:11]: I can't see where this is being used?

Comment (kenoir) [reply infrastructure/critical/locals.tf:11]: It's an [output](https://github.com/wellcomecollection/catalogue/pull/1261/files#diff-81d64fabe28d970892cf46ff6bc61d5b576d4e7cd2dfb6ceb16e7c5e985d29abR40).

Comment (jamesgorrie) [reply infrastructure/critical/locals.tf:11]: That it is - silly me.

Comment (kenoir) [reply infrastructure/critical/locals.tf:11]: Services will use that address as the URI for their Elastic Cloud deployment, as it routes requests via the VPC endpoint (and the deployment ID tells Elastic Cloud the other side which deployment to reference).

**Review by jamesgorrie — APPROVED** [review]
What do we need locally / in CI / wherever to have this work? I thought you said there was some env vars? Is it worth documenting them if so?

**Comment by kenoir** [comment]
> What do we need locally / in CI / wherever to have this work? I thought you said there was some env vars? Is it worth documenting them if so?

Good point. I haven't added that in this PR. I need to do that.

**Comment by kenoir** [comment]
This is now applied.

**Pull request merged**